### PR TITLE
BT-11899 check for kv cache reuse

### PR DESCRIPTION
**Pull request by bdubayah**

### CODE REVIEW

**Review by joostinyi — APPROVED** [review]

**Review by dsingal0 — APPROVED** [review]

File: truss/templates/trtllm-briton/src/engine.py [hunk 1]
```diff
@@ -90,6 +90,9 @@ def __init__(self, **kwargs):
         self._kv_cache_free_gpu_mem_fraction = (
             truss_trtllm_build_config.kv_cache_free_gpu_mem_fraction
         )
+        self._enable_kv_cache_reuse = (
+            truss_trtllm_build_config.plugin_configuration.use_paged_context_fmha
```
Comment (pankajroark) [inline truss/templates/trtllm-briton/src/engine.py:94]: I'm a bit confused. Does use_paged_context_fmha always mean kv cache reuse? I thought it was the other way. Is there a setting in our build config that directly indicates kv cache reuse and then everything could be governed by that.

**Review by pankajroark — APPROVED** [review]
lgtm, just one question around how enable kv cache use is defined in the config and used.

File: truss/templates/trtllm-briton/src/engine.py (hunk 1 above)
Comment (bdubayah) [reply truss/templates/trtllm-briton/src/engine.py:94]: `use_paged_context_fmha` must be enabled when the engine is built for kv cache reuse

But I agree that from the user perspective, they just want to set kv cache reuse and not worry about paged context fmha. @joostinyi @dsingal0 how about we land this and then follow up to just have kv cache reuse as an option in the builder config?

Comment (joostinyi) [reply truss/templates/trtllm-briton/src/engine.py:94]: Yeah that's true – existence of `use_paged_context_fmha` doesn't necessarily mean that `kv_cache_reuse` should be enabled. IMO there should probably be an explicit `trt_llm.runtime.enable_kv_cache_reuse` configuration that sets the `use_paged_context_fmha` config.

**Pull request merged**